File: src/com/digosofter/digowebapp/Objeto.java
package com.digosofter.digowebapp;

import com.digosofter.digowebapp.erro.Erro;

public abstract class Objeto {

  private static int _intIndex;

  private static int getIntIndex() {

    return _intIndex;
  }

  private static void setIntIndex(int intIndex) {

    _intIndex = intIndex;
  }

  private int _intId;

  private String _strNome;

  private String _strNomeExibicao;

  public Objeto() {

    try {

      Objeto.setIntIndex(Objeto.getIntIndex() + 1);
      this.setIntId(Objeto.getIntIndex());

    }
    catch (Exception ex) {

      new Erro("Erro inesperado.\n", ex);

    }
    finally {
    }
  }

  public int getIntId() {

    return _intId;
  }

  public String getStrNome() {

    return _strNome;
  }

  public String getStrNomeExibicao() {

    try {

      if (Utils.getBooStrVazia(_strNomeExibicao)) {

        if (!Utils.getBooStrVazia(this.getStrNome())) {

          _strNomeExibicao = Utils.getStrPrimeiraMaiuscula(this.getStrNome());
        }
      }

    }
    catch (Exception ex) {

      new Erro("Erro inesperado.\n", ex);

    }
    finally {
    }

    return _strNomeExibicao;
  }

  public String getStrNomeSimplificado() {

    return Utils.getStrSimplificada(this.getStrNome());
  }

  public void setIntId(int intId) {

    _intId = intId;
  }

  public void setStrNome(String strNome) {

    _strNome = strNome;
  }

  public void setStrNomeExibicao(String strNomeExibicao) {

    try {

      _strNomeExibicao = strNomeExibicao;
      _strNomeExibicao = Utils.getStrPrimeiraMaiuscula(_strNomeExibicao);

    }
    catch (Exception ex) {

      new Erro("Erro inesperado.\n", ex);

    }
    finally {
    }
  }

}

File: JDigo/Objeto.js
// CONSTANTE
// FIM CONSTANTE

// ATRIBUTOS GLOBAIS

var Objeto_intId = 0;

// FIM ATRIBUTOS GLOBAIS

// CLASSE
function Objeto() {
  // HERANÇA
  // FIM HERANÇA

  // ATRIBUTO

  var _this = this;

  var _intObjetoId = 0;

  this.getIntObjetoId = function() {
    return _intObjetoId;
  }

  this.setIntObjetoId = function(intObjetoId) {
    _intObjetoId = intObjetoId;
  }

  var _strNome = "";

  this.getStrNome = function() {
    return _strNome;
  }

  this.setStrNome = function(strNome) {
    _strNome = strNome;
  }

  var _strNomeExibicao = "";

  this.getStrNomeExibicao = function() {

    try {

      if (!Utils.getBooVazia(_strNomeExibicao)) {

        return _strNomeExibicao;
      }

      _strNomeExibicao = _this.getStrNome();

    } catch (e) {

      new Erro("Erro inesperado.", e);
    }

    return _strNomeExibicao;
  }

  this.setStrNomeExibicao = function(strNomeExibicao) {
    _strNomeExibicao = strNomeExibicao;
  }

  // FIM ATRIBUTO

  // MÉTODO
  // FIM MÉTODO

  /* Construtor */
  {

    try {

      Objeto_intId++;
      _this.setIntObjetoId(Objeto_intId);

    } catch (e) {
      new Erro("Erro inesperado.", e);
    }
  }
}